packages/react-celo/src/ethers.ts

```diff
@@ -6,6 +6,9 @@ import { Maybe } from './types';
 import { useCelo } from './use-celo';
 import { useIsMounted } from './utils/useIsMounted';
 
+// All the hooks in this file will created a new instance of whatever they memoize per component.
+// This isn't the best. Instead these should be using Context to ensure they all access the same object
+
```

Comment:
While you're correct, I'm actually wondering if this even used anywehre, the only function calling `useProvider` is `useProviderOrSigner`, itself being only used by `useGetConnectedSigner`, itself used by only `useLazyConnectedSigner` and `useConnectedSigner`. The 2 last function aren't being used anywhere AFAIK.